### PR TITLE
Update workflows

### PR DESCRIPTION
**Pull request by lpusok**

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -1,903 +1,53 @@
...
-        echo "GIT_CLONE_COMMIT_AUTHOR_EMAIL: ${GIT_CLONE_COMMIT_AUTHOR_EMAIL}"
-        echo "GIT_CLONE_COMMIT_COMMITER_NAME: ${GIT_CLONE_COMMIT_COMMITER_NAME}"
-        echo "GIT_CLONE_COMMIT_COMMITER_EMAIL: ${GIT_CLONE_COMMIT_COMMITER_EMAIL}"
-      opts:
-        is_expand: false
-    # define these envs in your .bitrise.secrets.yml
-    - GIT_CLONE_SSH_PRIVATE_KEY: $GIT_CLONE_SSH_PRIVATE_KEY
+    - TEST_APP_URL: https://github.com/bitrise-io/git-clone-test.git
+    - TEST_APP_BRANCH: master
+    - SHOULD_ACTIVATE_SSH_KEY: true # Set to true to test with private repository
```
Comment (mateherber) [inline bitrise.yml:8]: I wouldn't add configurability to the sample workflow. The test app URL is public so we don't need auth support. If you want to test it you should edit the sample wf as you see fit or use the e2e steps.

File: bitrise.yml [hunk 2]
```diff
@@ -1,903 +1,53 @@
...
+            - clone_into_dir: .
+            - repository_url: $TEST_APP_URL
+            - branch: $TEST_APP_BRANCH
       - script:
           inputs:
             - content: |-
                 #!/bin/env bash
-                set -ex
-                stepman audit --step-yml ./step.yml
+                echo "GIT_CLONE_COMMIT_HASH: ${GIT_CLONE_COMMIT_HASH}"
```
Comment (mateherber) [inline bitrise.yml:45]: I wouldn't add this "validation" script either

File: e2e/bitrise.yml [hunk 3]
```diff
@@ -0,0 +1,688 @@
...
+            - repository_url: $TEST_REPO_URL
+            - clone_into_dir: $WORKDIR
+            - tag: $BITRISE_GIT_TAG
+            - branch: "test/generate-changelog"
+            - fetch_tags: "yes"
+            - update_submodules: "no"
+    after_run:
+      - _assert_changelog
+
+  _assert_changelog:
```
Comment (mateherber) [inline e2e/bitrise.yml:56]: I think this is a `_check_*` wf

File: e2e/bitrise.yml [hunk 4]
```diff
@@ -0,0 +1,688 @@
...
+                envman unset --key GIT_CLONE_COMMIT_HASH
+                envman unset --key GIT_CLONE_COMMIT_MESSAGE_SUBJECT
+                envman unset --key GIT_CLONE_COMMIT_MESSAGE_BODY
+                envman unset --key GIT_CLONE_COMMIT_COUNT
+                envman unset --key GIT_CLONE_COMMIT_AUTHOR_NAME
+                envman unset --key GIT_CLONE_COMMIT_AUTHOR_EMAIL
+                envman unset --key GIT_CLONE_COMMIT_COMMITER_NAME
+                envman unset --key GIT_CLONE_COMMIT_COMMITER_EMAIL
+
+  _log_outputs:
```
Comment (mateherber) [inline e2e/bitrise.yml:661]: This is also a `_check_*` wf

File: e2e/bitrise.yml [hunk 5]
```diff
@@ -0,0 +1,688 @@
...
+                fi
+                if [ ${#GIT_CLONE_COMMIT_MESSAGE_SUBJECT} -gt "20480" ]; then
+                  echo "Message subject is bigger than 20480 chars"
+                  exit 1
+                fi
+    after_run:
+      - _log_outputs
+      - _ensure_clean_git
+
+  _prepare:
```
Comment (mateherber) [inline e2e/bitrise.yml:623]: Probably we should have a naming convention for the `setup` kind of workflows too like `prepare` now. We might also have `teardown` kind of wfs like `_ensure_clean_git` now

I'm thinking like

```
test_*                  Test workflows
utility_*               Workflows that are used from tests
_check_*             Verify or print results
_setup_*             Prepare state for test
_teardown_*      Clean up state after test
```

**Review by mateherber — CHANGES_REQUESTED** [review]

**Comment by lpusok** [comment]
> I wouldn't add this "validation" script either 

removed.

**Comment by lpusok** [comment]
> This is also a `_check_*` wf 

Updated.

File: e2e/bitrise.yml (hunk 5 above)
Comment (mateherber) [reply e2e/bitrise.yml]: You're right. Then I think we don't need the _setup/_teardown wfs just the _run but we will have to refactor this yml to that.

```
test_*                  Test workflows (prepare env vars for _run basically)
utility_*               Workflows that are used from tests
_check_*             Verify or print results
_run                    3 phase of step execution: setup, run the step, teardown
```

File: e2e/bitrise.yml [hunk 6]
```diff
@@ -0,0 +1,688 @@
...
+          inputs:
+            - repository_url: $TEST_REPO_URL
+            - clone_into_dir: $WORKDIR
+            - commit: "76a934ae80f12bb9b504bbc86f64a1d310e5db64"
+            - branch: "test/commit-messages"
+            - update_submodules: "no"
+      - script:
+          inputs:
+            - content: |-
+                #!/usr/bin/env bash
```
Comment (mateherber) [inline e2e/bitrise.yml:395]: this should be extracted as check wf

**Review by istvankovacs-bitrise — APPROVED** [review]

**Review by mateherber — APPROVED** [review]

**Pull request merged**